### PR TITLE
[19.07] kirkwood: increase kernel partition of Linksyses

### PR DESCRIPTION
**Pull request by CHKDSK88**
At this moment kernel partition in Linksyses EA3500/E4200/EA4500 is
ended before start of rootfs partition. It was introduced in 9808b9ae02
and it broke easy revert to stock. Sysupgrade, when OFW is used, write
whole stock image to kernel partition. Most likeley image will be bigger
than small kernel partition and it make stock system invalid.

This patch change size of kernel partitions and now it overlaps rootfs.
Revert to stock will be possible again.

Fixes: 9808b9ae02 ("kirkwood: switch to kernel 4.9")

Signed-off-by: Pawel Dembicki <paweldembicki@gmail.com>

### CODE REVIEW

**Review by ynezz — APPROVED** [review]
LGTM, but I would like @mkresin opinion as well before merging it that late in 19.07 release cycle.

**Comment by mkresin** [comment]
//cc @hauke 

I really don't like this change, neither do I like the similar change in master (0dbe754e4328e1ade02b29f25f0ebc633305d5bb).

With these changes, we now have overlapping partitions (kernel1 covers rootfs1 as well), which always looks like a mistake and makes it hard to get how much space for the kernel is really available. Furthermore, it's potential source for a brick. Considering a kernel which is that big that it covers part of the rootfs partition as well. As soon as the rootfs is written to its partition the remains of the kernel image are overwritten and the kernel image is corrupted.

And all that for having a hack in place which allows for an easy revert to the OEM firmware from a running OpenWrt.

From my point of view, such a hack shouldn't be backported, especially since it's not even remotely a bugfix (for an OpenWrt issue).

However, I'm not going to block this change. Due to the check in the image build code, which ensures the kernel size is limited to 0x290000/0x2a0000 bytes, it shouldn't break anything.

**Comment by ynezz** [comment]
> Furthermore, it's potential source for a brick.

Yes, I've same concerns and since 19.07 is going to be EOLed soon and should be really only about security updates, I'm not going to merge it, sorry about that @CHKDSK88, but thank you for your efforts.

> I really don't like this change, neither do I like the similar change in master (https://github.com/openwrt/openwrt/commit/0dbe754e4328e1ade02b29f25f0ebc633305d5bb).

FYI it's already in 21.02 as well 0e29e05deed1edd3884ca888ae7e1b549c84bfd0

**Comment by CHKDSK88** [comment]
@ynezz 

When it was backported to 21.02, this PR became less important for me. 

P.S.
I'm understood that is ugly solution. But for regular user, this ugly solution is better than brick if revert to stock was unsuccessful. Maybe in future we can make "legacy" partition, which overlaps 'kernel' and 'ubi'. 

After https://github.com/openwrt/openwrt/commit/0dbe754e4328e1ade02b29f25f0ebc633305d5bb, easy revert to stock is very important for update to snapshot or future releases >=22.X.

I'm going to describe on wiki full instruction how to use factory image for sysupgrade from 1.0 to 2.0 images.

**Pull request closed without merging**
